zulipterminal/ui_tools/buttons.py

```diff
@@ -86,6 +87,10 @@ def activate(self, key: Any) -> None:
         self.controller.view.show_left_panel(visible=False)
         self.controller.view.show_right_panel(visible=False)
         self.controller.view.body.focus_col = 1
+        self.set_status(is_active=True)
+        if self.model.active_button:
+            self.model.active_button.set_status(is_active=False)
+        self.model.active_button = self
```

Comment:
My understanding is that this code is only triggering when activating a focused button via eg. space or enter (or clicking), not when using keys to otherwise change the narrow - when the highlights also need to change. So, while this logic works (with some issues right now?), it will also end up being a duplicate of the logic we really want.

Reply:
Right now it's changed to work only for streams, so I think this is no longer a concern? I might be wrong.